extensions/azdata/src/test/azdata.test.ts

```diff
@@ -208,50 +213,41 @@ async function testDarwinSuccessfulUpdate() {
 	}];
 	const executeCommandStub = sinon.stub(childProcess, 'executeCommand')
 		.onThirdCall() //third call is brew info azdata-cli --json which needs to return json of new available azdata versions.
-		.callsFake(async (command: string, args: string[]) => {
-			should(command).be.equal('brew');
-			should(args).deepEqual(['info', 'azdata-cli', '--json']);
-			return Promise.resolve({
-				stderr: '',
-				stdout: JSON.stringify(brewInfoOutput)
-			});
+		.resolves({
+			stderr: '',
+			stdout: JSON.stringify(brewInfoOutput)
 		})
-		.callsFake(async (_command: string, _args: string[]) => { // return success on all other command executions
-			return Promise.resolve({ stdout: '0.0.0', stderr: '' });
-		});
+		.resolves({ stdout: '0.0.0', stderr: '' });
 	await azdata.checkAndUpdateAzdata(oldAzdataMock);
 	should(executeCommandStub.callCount).be.equal(6);
+	should(executeCommandStub.getCall(2).args[0]).be.equal('brew', '3rd call should have been to brew');
+	should(executeCommandStub.getCall(2).args[1]).deepEqual(['info', 'azdata-cli', '--json'], '3rd call did not have expected arguments');
 }
 
 
 async function testWin32SuccessfulUpdate() {
 	sinon.stub(HttpClient, 'getTextContent').returns(Promise.resolve(JSON.stringify(releaseJson)));
 	sinon.stub(HttpClient, 'downloadFile').returns(Promise.resolve(__filename));
-	const executeCommandStub = sinon.stub(childProcess, 'executeCommand').callsFake(async (command: string, args: string[]) => {
-		should(command).be.equal('msiexec');
-		should(args[0]).be.equal('/qn');
-		should(args[1]).be.equal('/i');
-		return { stdout: '0.0.0', stderr: '' };
-	});
 	await azdata.checkAndUpdateAzdata(oldAzdataMock);
-	should(executeCommandStub.calledOnce).be.true();
+	should(executeSudoCommandStub.calledOnce).be.true('executeSudoCommand should have been called once');
+	should(executeSudoCommandStub.getCall(0).args[0]).startWith('msiexec /qn /i');
 }
 
 async function testWin32SuccessfulInstall() {
+	sinon.stub(HttpClient, 'getTextContent').returns(Promise.resolve(JSON.stringify(releaseJson)));
 	sinon.stub(HttpClient, 'downloadFile').returns(Promise.resolve(__filename));
 	const executeCommandStub = sinon.stub(childProcess, 'executeCommand')
 		.onFirstCall()
-		.callsFake(async (_command: string, _args: string[]) => {
-			return Promise.reject(new Error('not Found'));
-		})
-		.callsFake(async (command: string, args: string[]) => {
-			should(command).be.equal('msiexec');
-			should(args[0]).be.equal('/qn');
-			should(args[1]).be.equal('/i');
-			return { stdout: '0.0.0', stderr: '' };
+		.rejects(new Error('not Found')) // First call mock the tool not being found
+		.resolves({ stdout: '1.0.0', stderr: '' });
+	executeSudoCommandStub
+		.callsFake(async (command: string) => {
+			should(command).startWith('msiexec');
```

Comment:
nit: convert this also to 	should(executeSudoCommandStub.getCall(0).args[0]).startWith('msiexec /qn /i'); and change the callsFage to resolves for consistency

Reply:
Good catch. And FYI - the reason I updated this was because the way the logic was now this check wasn't failing the test if it failed. They use errors for doing that and we were catching those errors so it wasn't bubbling up properly. Moving it out to be done at the test level fixed that issue.